### PR TITLE
Reality Check - Show Page

### PR DESCRIPTION
**Pull request by zach-bergman**
required_monthly_income method and test - in Facade.
reality_check action in the ListingsController - I don't like this, want to refactor eventually.
bootstrap modal in show view, javascript added to show modal on show view once monthly income is entered

### CODE REVIEW

**Comment by jdmchugh111** [comment]
should we wait until @RodrigoACG approves and merges my FE PR from last night so we can check against yours for merge conflicts?

File: app/views/listings/show.html.erb [hunk 1]
```diff
@@ -42,8 +42,44 @@
...
+        </div>
+        <%= form.submit "Submit", class: "btn btn-primary" %>
+      <% end %>
+    </div>
+  </div>
+
+  <!-- Modal -->
+  <div class="modal fade" id="resultModal" tabindex="-1" role="dialog" aria-labelledby="resultModalLabel" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+      <div class="modal-content">
```
Comment (RodrigoACG) [inline app/views/listings/show.html.erb:69]: All of this looks amazing good job once again Zach!!

**Review by RodrigoACG — APPROVED** [review]

File: config/routes.rb [hunk 2]
```diff
@@ -17,5 +17,7 @@
   
   resources :reality_check, only: [:index]
 
-  resources :listings, only: [:show]
+  resources :listings, only: [:show] do
+    post 'reality_check', on: :member
+  end
```
Comment (jdmchugh111) [inline config/routes.rb:23]: what does the on: :member syntax do?

File: app/views/listings/show.html.erb [hunk 3]
```diff
@@ -42,8 +42,44 @@
...
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          <p>What You Make: <%= number_to_currency(cookies[:monthly]) %></p>
+          <p>What You Would Need To Make: <%= number_to_currency(cookies[:required_income]) %></p>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+        </div>
```
Comment (jdmchugh111) [inline app/views/listings/show.html.erb:82]: first time seeing use of a modal in a view, but it looks good!

**Pull request merged**